### DISCUSSION TITLE
Unsupported engine for redis@4.0.2

### DISCUSSION THREAD

**Discussion by youngha-sung** [renovatebot/renovate#13787] [Request Help]
### How are you running Renovate?

Self-hosted

### Please select which platform you are using if self-hosting.

Azure DevOps Server

### Describe the bug

We have been using renovate with node and npm version specified in package.json as below and it has been working until the last week. 

```
  "engines": {
    "node": "14.x",
    "npm": "6.x"
  },
```

From last week we started to get the following error when we run `npx --userconfig .npmrc renovate`: 

```
Starting: Bash
==============================================================================
Task         : Bash
Description  : Run a Bash script on macOS, Linux, or Windows
Version      : 3.198.0
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/bash
==============================================================================
Generating script.
========================== Starting Command Output ===========================
/usr/bin/bash /home/vsts/work/_temp/f1803e60-a115-4946-af9e-6748dcfa619a.sh
npm ERR! code ENOTSUP
npm ERR! notsup Unsupported engine for redis@4.0.2: wanted: {"npm":">=7","typescript":">=4"} (current: {"node":"14.18.3","npm":"6.14.15"})
npm ERR! notsup Not compatible with your version of node/npm: redis@4.0.2
npm ERR! notsup Not compatible with your version of node/npm: redis@4.0.2
npm ERR! notsup Required: {"npm":">=7","typescript":">=4"}
npm ERR! notsup Actual:   {"npm":"6.14.15","node":"14.18.3"}

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/vsts/.npm/_logs/2022-01-24T05_01_05_349Z-debug.log
Install for [ 'renovate@latest' ] failed with code 1
##[error]Bash exited with code '1'.
Finishing: Bash
```

We have not configured `redisUrl` anywhere so not sure why we are getting this redis error. 

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste any log here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by viceice** [comment]
You need to update the global npm version. Run `npm i -g npm` on ci, and upgrade your npm engine value to be `7.x` or higher.

**Reply by joelgrimberg** [reply]
after installing npm -g npm, still the npm from the docker-image is being used.

**Comment by youngha-sung** [comment]
@viceice Is it not backward compatible? Seems weird that our pipeline is broken because of the redis error when we are not enabling the redis. We can upgrade NPM to `7.x` or higher but wondering if any quick fix is available as well, as I suspect there are many users having similar issue?

**Comment by viceice** [comment]
@rarkins i  think we can make Redis optional like re2 and only throw error when Redis url is set and Redis is missing?
